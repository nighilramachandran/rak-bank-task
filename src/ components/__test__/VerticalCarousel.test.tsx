import { render, screen } from "@testing-library/react";
import VerticalCarousel, { carouslProps } from "../vertical-carousel";

const setup = (props: Partial<carouslProps> = {}) => {
  const defaultProps: carouslProps = {
    slides: [
      { description: "Question 1", polling: <div>Emot Slide 1</div> },
      { description: "Question 2", polling: <div>Emot Slide 2</div> },
    ],
    currentSlide: 0,
    type: "question",
  };
  const mergedProps = { ...defaultProps, ...props };

  render(<VerticalCarousel {...mergedProps} />);
};

describe("VerticalCarousel component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders question slide with description", () => {
    setup();
    const text = screen.getByTestId("type-question-index-0").textContent;
    expect(text).toBe("Question 1");
    expect(text).not.toBe("Question 2");
    expect(text).not.toBe("Emot Slide 1");
  });

  test("renders emot slide with polling content", () => {
    setup({ type: "emot" });
    const text = screen.getByTestId("type-emot-index-0").textContent;
    expect(text).toBe("Emot Slide 1");
    expect(text).not.toBe("Emot Slide 2");
    expect(text).not.toBe("Question 1");
  });
});
