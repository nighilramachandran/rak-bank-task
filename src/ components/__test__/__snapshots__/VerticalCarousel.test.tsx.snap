// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalCarousel component match the snapshot 1`] = `
<body>
  <div>
    <div
      style="opacity: 1; transform: none;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pb4qw9-MuiTypography-root"
        data-testid="type-question-index-0"
      >
        Question 1
      </p>
    </div>
  </div>
</body>
`;
